mods/ra/lua/ai.lua

```diff
@@ -0,0 +1,1072 @@
+			best_a = a
+			best_cnt = #stuff
+		end
+	end
+	--Media.DisplayMessage("least protected OK", "lua_ai")
+	return best_a
+end
+
+UTIL_count_alive = function(actors)
+	local alive = 0
+	for _, a in ipairs(actors) do
+		if not a.IsDead then
+			alive = alive + 1
+		end
+	end
+	return alive
+end
+
+UTIL_wait_load = function(transport, passengers, next_task, params_next_task)
+	if transport.IsDead then
+		TASK_unoccupy(passengers)
+	end
+
+	if transport.PassengerCount >= UTIL_count_alive(passengers) then
+		next_task(params_next_task)
+	else
+		Trigger.AfterDelay(30, function()
+			-- invoke self again to implement "wait load".
+			UTIL_wait_load(transport, passengers, next_task, params_next_task)
+		end)
+	end
+end
+
+UTIL_load_passengers = function(transport, passengers, next_task, params_next_task)
+	-- Order just once, or else it will bug.
+	for _, a in ipairs(passengers) do
+		a.Wait(45) -- wait until transport is out of the factory.
+		a.EnterTransport(transport)
+	end
+	UTIL_wait_load(transport, passengers, next_task, params_next_task)
+end
+
+MOD_is_anypower = function(name)
+	if name == "powr" then
+		return true
+	elseif name == "apwr" then
+		return true
+	end
+	return false
+end
+
+--
+-- Given faction, choose a build order.
+-- Currently, random.
+--
+MOD_choose_build_order = function(faction)
+	local build_order = nil
+
+	if faction == "soviet" or faction == "ukraine" or faction == "russia" then
+		build_order = BO_SOVIET_BOS
+	elseif faction == "allies" or faction == "england" or faction == "france" or faction == "germany" then
+		build_order = BO_ALLIES_BOS
+	else
+		-- some mods may be under development.
+		BUILD_ORDER = nil
+		return
+	end
+
+	BUILD_ORDER = Utils.Random(build_order)
+	--BUILD_ORDER = BO_SOVIET_FLAME_OPEN -- lets watch
+
+	-- select opener so it may be ticked.
+	UB_OPENER_TICK = Utils.Random(BUILD_ORDER.openers)
+end
+
+MOD_is_soviet = function(faction_name)
+  if faction_name == "soviet" or faction_name == "ukraine" or faction_name == "soviet" then
+    return true
+  else
+    return false
+  end
+end
+
+MOD_is_allies = function(faction_name)
+  if faction_name == "allies" or faction_name == "england" or faction_name == "germany" or faction_name == "france" then
+    return true
+  else
+    return false
+  end
+end
+
+TASKFORCES =
+{
+	-- common stuff
+	["mcv"] = {"mcv"},
+	["harv"] = {"harv"},
+	["1e6"] = {"e6"},
+
+	-- Allied taskforces.
+	-- These are basic units of production. They may be merged into a big team.
+	["6heli"] = {"heli", "heli", "heli", "heli", "heli", "heli"},
+	["2hind"] = {"hind", "hind"},
+	["10e1"] = {"e1", "e1", "e1", "e1", "e1", "e1", "e1", "e1", "e1", "e1"},
+	["5e1"] = {"e1","e1","e1","e1","e1"},
+	["5e3"] = {"e3", "e3", "e3", "e3", "e3"},
+	["tany1"] = {"e7", "jeep"}, -- ride and bomb
+	["tany2"] = {"e7", "stnk"}, -- STEALTH yeah
+	["2jeep"] =  {"jeep", "jeep"},
+	["4x1tnk"] = {"1tnk", "1tnk", "1tnk", "1tnk"},
+	["4x2tnk"] = {"2tnk", "2tnk", "2tnk", "2tnk"},
```

Comment:
Then maybe I should consider using underscore (_) than x, some units (not in RA mod though) may start with an x. Didn't use x when unit name doesn't start with a number.